docs/architecture/architecture.md

```diff
@@ -14,6 +14,91 @@ built to be multi-tenant.
 | [Private SaaS (Customer VPC)](./deployment#private-saas-customer-vpc) | Enterprises in the Cloud                                        |
 | [On-Premise](./deployment#on-premise-deployment)                      | Large Enterprises in the middle of cloud migration (rare cases) |
 
+## High-Level Architecture
+
+There are four components of a successful Prophecy deployment:
+
+- **Prophecy IDE** - The development environment, including Prophecy Microservices and Cloud Infrastructure, which is deployed using one of these options: [Public SaaS](#public-saas), [Private SaaS](#private-saas-customer-vpc), or (rarely) [On-Premise](#on-premise-deployment).
+
+- **Data Engine (eg Snowflake or Databricks)** - [SQL](#sql) or [Spark](#spark) execution environment. This is setup by a customer and connected to Prophecy through a secure and performant interface. No customer’s data is stored on Prophecy’s environment.
+
+- **Source Control (e.g. Bitbucket)** - Prophecy works similar to code-first IDEs, by natively integrating with [Git](#git) for best software engineering practices. An encrypted copy of customer’s code is stored within Prophecy’s environment for fast access, while the source-of-truth code is stored on Git.
+
+- **Identity Management (optional)** - For simple user authentication and permission control, Prophecy can connect your identity provider of choice.
```

Comment:
The four components maybe outdated now. I guess there are a lot more core components like metagraph, metadata-ui and additional services like platform (Athena). And even optional services like copilot, transpiler etc. 

Not sure if we want to revisit this now or later.

Reply:
Probably on high level these 4 are  basic requirements , Lets keep them for now